pkg/scheduler/core/division_algorithm.go

```diff
@@ -0,0 +1,205 @@
+				weightSum += staticWeightRule.Weight
+				matchClusters[cluster.Name] = staticWeightRule.Weight
+				break
+			}
+		}
+	}
+
+	if weightSum == 0 {
+		for _, cluster := range clusters {
+			weightSum++
+			matchClusters[cluster.Name] = 1
+		}
+	}
+
+	allocatedReplicas := int32(0)
+	for clusterName, weight := range matchClusters {
+		desireReplicaInfos[clusterName] = weight * int64(replicas) / weightSum
+		allocatedReplicas += int32(desireReplicaInfos[clusterName])
+	}
+
+	if remainReplicas := replicas - allocatedReplicas; remainReplicas > 0 {
+		sortedClusters := helper.SortClusterByWeight(matchClusters)
+		for i := 0; remainReplicas > 0; i++ {
+			desireReplicaInfos[sortedClusters[i].ClusterName]++
+			remainReplicas--
+			if i == len(desireReplicaInfos) {
+				i = 0
+			}
+		}
+	}
+
+	for _, cluster := range clusters {
+		if _, exist := matchClusters[cluster.Name]; !exist {
+			desireReplicaInfos[cluster.Name] = 0
+		}
+	}
+
+	targetClusters := make([]workv1alpha2.TargetCluster, len(desireReplicaInfos))
+	i := 0
+	for key, value := range desireReplicaInfos {
+		targetClusters[i] = workv1alpha2.TargetCluster{Name: key, Replicas: int32(value)}
+		i++
+	}
+	return targetClusters, nil
+}
+
+// divideReplicasByPreference assigns a total number of replicas to the selected clusters by preference according to the resource.
+func divideReplicasByPreference(clusterAvailableReplicas []workv1alpha2.TargetCluster, replicas int32, clustersMaxReplicas int32,
+	preference policyv1alpha1.ReplicaDivisionPreference, preUsedClustersName ...string) ([]workv1alpha2.TargetCluster, error) {
+	switch preference {
+	case policyv1alpha1.ReplicaDivisionPreferenceAggregated:
+		return divideReplicasByAggregation(clusterAvailableReplicas, replicas, preUsedClustersName...), nil
+	case policyv1alpha1.ReplicaDivisionPreferenceWeighted:
+		return divideReplicasByAvailableReplica(clusterAvailableReplicas, replicas, clustersMaxReplicas), nil
+	default:
+		return nil, fmt.Errorf("undefined replicaSchedulingType： %v", preference)
+	}
+}
+
+func divideReplicasByAggregation(clusterAvailableReplicas []workv1alpha2.TargetCluster,
+	replicas int32, preUsedClustersName ...string) []workv1alpha2.TargetCluster {
+	clusterAvailableReplicas = presortClusterList(clusterAvailableReplicas, preUsedClustersName...)
+	clustersNum, clustersMaxReplicas := 0, int32(0)
+	for _, clusterInfo := range clusterAvailableReplicas {
+		clustersNum++
+		clustersMaxReplicas += clusterInfo.Replicas
+		if clustersMaxReplicas >= replicas {
+			break
+		}
+	}
+	var unusedClusters []string
+	for i := clustersNum; i < len(clusterAvailableReplicas); i++ {
+		unusedClusters = append(unusedClusters, clusterAvailableReplicas[i].Name)
+	}
+	return divideReplicasByAvailableReplica(clusterAvailableReplicas[0:clustersNum], replicas, clustersMaxReplicas, unusedClusters...)
+}
+
+func divideReplicasByAvailableReplica(clusterAvailableReplicas []workv1alpha2.TargetCluster, replicas int32,
```

Comment:
I wonder if we can reuse some logic from `static weight division`? If so, here we can focus on making a weight list according to available replicas?

Reply:
Maybe not a good idea... It's too complicated because we should consider merging the second-schedule result to the previously scheduled result.